proj/src/main/webapp/signup.js

```diff
@@ -143,15 +164,35 @@ function initializePasswordValidation(){
             passconf.classList.add("non-matching-pass");
             noMatch.hidden = false;
         }
-        if (meetRequirements(passVal) && noMatch.hidden) {
-            btnSignUp.disabled = false;
-        } else {
-            // disable sign up button
-            btnSignUp.disabled = true;
-        }
+        enableSignUp(passVal);
     })
 }
 
+function hasTag(){
+    const tagBox = document.getElementById("tags");
+    if (tagBox.value.trim().length != 0){
+        $('#tags').css('border','none');
+        return true;
+    } else {
+        // the tag box will only be in red after the user has input a valid password 
+        // this is to prevent the page from being visually overwhelming and help guide the user
+        if (noMatch.hidden){
```

Comment:
I don't think noMatch is defined in a scope where this function can see it. You might need to re-find noMatch in the DOM if you want to use it here.

Reply:
you are right, it seems like it is not defined in the `hasTag` function but it does work.  for security, i will declare it within the scope